system-int-test/test/cmr/system_int_test/search/scrolling_search_test.clj

```diff
@@ -0,0 +1,149 @@
+    ;; The following test should not be removed and only uncommented during manual testing.
+    
+    #_(testing "Expired scroll-id is invalid"
+        (let [timeout (es-config/elastic-scroll-timeout)
+              ;; Set the timeout to one second
+              _ (es-config/set-elastic-scroll-timeout! "1s")
+              {:keys [scroll-id] :as result} (search/find-refs 
+                                              :granule 
+                                              {:provider "PROV1" :scroll true :page-size 2})]   
+          (testing "First call returns scroll-id and hits count with page-size results"
+            (is (data2-core/refs-match? [gran1 gran2] result)))  
+          ;; This is problematic. Can't use timekeeper tricks here since ES is the one enforcing 
+          ;; the timeout, and it seems to have it's own scheule, so our session id does not time out
+          ;; in exactly one second. 
+          (Thread/sleep 100000)
+          (testing "Subsequent calls get unknown scroll-id error"
+            (let [response (search/find-refs :granule 
+                                            {:scroll true}
+                                            {:allow-failure? true
+                                             :headers {routes/SCROLL_ID_HEADER scroll-id}})]
+              (is (= 404 (:status response)))
+              (is (= (str "Scroll session [" scroll-id "] does not exist")
+                    (first (:errors response))))))
+          (es-config/set-elastic-scroll-timeout! timeout)))
+
+    (testing "invalid parameters"
+      (are3 [query options status err-msg]
+        (let [options (merge {:allow-failure? true})
+              response (search/find-refs :granule query options)]
+          (is (= status (:status response)))
+          (is (= err-msg
+                 (first (:errors response)))))
+
+        "Scroll queries cannot be all-granule queries"
+        {:scroll true}
+        {}
+        400
+        (str "The CMR does not allow querying across granules in all collections when scrolling." 
+             " You should limit your query using conditions that identify one or more collections "
+             "such as provider, concept_id, short_name, or entry_title.")
+
+        "scroll parameter must be boolean"
+        {:provider "PROV1" :scroll "foo"} 
+        {}
+        400
+        "Parameter scroll must take value of true or false but was [foo]"
+
+        "page_num is not allowed with scrolling"
+        {:provider "PROV1" :scroll true :page-num 2}
+        {}
+        400
+        "page_num is not allowed with scrolling"
+        
+        "offset is not allowed with scrolling"
+        {:provider "PROV1" :scroll true :offset 2}
+        {}
+        400
+        "offset is not allowed with scrolling"
+
+        "Unknown scroll-id"
+        {:scroll true}
+        {:headers {routes/SCROLL_ID_HEADER "foo"}}
+        404
+        "Scroll session [foo] does not exist"))))
```

Comment:
We should add one test for the disabling configuration returning a 400 and an error message:
```
    (dev-sys-util/eval-in-dev-sys
     `(cmr.common-app.services.search.parameter-validation/set-scrolling-enabled! false))
```
Just remember to set it back to true after the test.